includes/components/class-attribute.php

```diff
@@ -950,22 +1042,30 @@ public function update_model_snippet( $model_id, $model ) {
 	 */
 	public function add_edit_fields( $form_args, $form ) {
 
+		// Get model name.
+		$model_name = $form::get_meta( 'model' );
+
 		// Get model.
-		$model = $form::get_meta( 'model' );
+		$model = $form->get_model();
 
 		// Get category IDs.
-		$category_ids = $this->get_category_ids( $model, $form->get_model() );
+		$category_ids = $this->get_category_ids( $model_name, $model );
 
 		// Get attributes.
-		$attributes = $this->get_attributes( $model, $category_ids );
+		$attributes = $this->get_attributes( $model_name, $category_ids );
 
 		foreach ( $attributes as $attribute_name => $attribute ) {
 			if ( $attribute['editable'] && ! isset( $form_args['fields'][ $attribute_name ] ) ) {
 
 				// Get field arguments.
 				$field_args = $attribute['edit_field'];
 
-				if ( $attribute['moderated'] && $model . '_update' === $form::get_meta( 'name' ) ) {
+				// Add readonly option.
+				if ( $attribute['readonly'] && call_user_func( [ $model, 'get_' . $attribute_name ] ) ) {
```

Comment:
Make sure that this check for an empty attribute value also allows zero numbers, maybe is_null will be ok (we use null as "not set" status for Fields)